contracts/eosio.system/src/eosio.system.cpp

```diff
@@ -397,6 +397,88 @@ namespace eosiosystem {
       set_resource_limits( account, current_ram, current_net, cpu );
    }
 
+   // --------------------------------------------------------------------------
+   struct blacklist_manager {
+      explicit blacklist_manager(std::vector<name>& blacklist) : _blacklist(blacklist) {}
+
+      void add(const std::vector<name>& patterns) const {
+         std::unordered_set<name, _hash> already(std::cbegin(_blacklist), std::cend(_blacklist));
+
+         for (auto n : patterns) {
```

Comment:
If `patterns` contain duplicated patterns, will they be pushed into _blacklist multiple times, as `already` was computed before?

Reply:
Good catch, that was fixed in https://github.com/eosnetworkfoundation/eos-system-contracts/pull/189/commits/f31f3cf29ced5bf3bed259701480aec39fe176d1.